pkgs/games/openra/build-engine.nix

```diff
@@ -7,28 +7,51 @@
   freetype,
   openal,
   lua51Packages,
+  openRaUpdater,
 }:
 engine:
 
-buildDotnetModule rec {
+let
   pname = "openra-${engine.build}";
-  inherit (engine) version;
-
-  src =
-    if engine ? src then
-      engine.src
-    else
-      fetchFromGitHub {
-        owner = "OpenRA";
-        repo = "OpenRA";
-        rev = "${engine.build}-${engine.version}";
-        sha256 = engine.sha256;
-      };
+  version = engine.version;
+
+  dotnetVersions = {
+    "6" = {
+      sdk = dotnetCorePackages.sdk_6_0-bin;
+      runtime = dotnetCorePackages.runtime_6_0-bin;
+    };
+    "8" = {
+      sdk = dotnetCorePackages.sdk_8_0;
+      runtime = dotnetCorePackages.runtime_8_0;
+    };
+  };
 
-  nugetDeps = engine.deps;
+  dotnet-sdk = lib.getAttrFromPath [ engine.dotnetVersion "sdk" ] dotnetVersions;
+  dotnet-runtime = lib.getAttrFromPath [ engine.dotnetVersion "runtime" ] dotnetVersions;
```

Comment:
Not a big deal, but each sdk has a default runtime that's used as a default for `dotnet-runtime`, and they are the same ones you have listed in `dotnetVersions` above, so maybe you don't need to specify it at all?

Reply:
good point, updated